PL2/PL2-gui/src/main/java/nl/tudelft/pl2016gr2/gui/model/PhylogeneticTreeNode.java

```diff
@@ -1,109 +1,240 @@
 package nl.tudelft.pl2016gr2.gui.model;
 
+import javafx.beans.property.BooleanProperty;
+import javafx.beans.property.SimpleBooleanProperty;
+import net.sourceforge.olduvai.treejuxtaposer.drawer.TreeNode;
+
 import java.util.ArrayList;
-import java.util.Objects;
+import java.util.Iterator;
+import java.util.NoSuchElementException;
 
 /**
- * This class is an adapter class which maps the methods of the
- * net.sourceforge.olduvai.treejuxtaposer.drawer.TreeNode class to the interface which is needed by
- * our application.
+ * This class is a storage class for a phylogenetic tree node. It retrieves and stores the needed
+ * values of the newick parser {@link TreeNode}.
  *
  * @author Faris
  */
-public class PhylogeneticTreeNode implements IPhylogeneticTreeNode {
+public class PhylogeneticTreeNode implements IPhylogeneticTreeNode, Iterable<PhylogeneticTreeNode> {
+
+  private final String label;
+  private final float weight;
+  private final PhylogeneticTreeNode[] children;
+  private final PhylogeneticTreeNode parent;
```

Comment:
Edit: the UI actually also stores the original root, but finding the parent of a node in the tree by starting at the root requires `O(n)` time, since in the worst case you will have to browse through all nodes to find the current node and then return its parent. Like this it is `O(1)`.